src/cpp/statistics/rtps/monitor-service/MonitorService.cpp

```diff
@@ -323,43 +302,29 @@ bool MonitorService::write_status(
 
                         break;
                     }
-                    case INCOMPATIBLE_QOS:
+                    case StatusKind::INCOMPATIBLE_QOS:
                     {
-                        rtps::DDSEntityStatus* dds_entity_status = new rtps::DDSEntityStatus;
-                        status_queryable_.get_monitoring_status(local_entity_guid, INCOMPATIBLE_QOS, dds_entity_status);
-
-                        assert(nullptr != dds_entity_status);
-
-                        IncompatibleQoSStatus_s incompatible_qos_status;
-                        incompatible_qos_status.policies().resize(dds_entity_status->policies.size());
-                        dds::QosPolicyCountSeq* qos_policy_countseq = &dds_entity_status->policies;
-                        incompatible_qos_status.policies() =
-                                *reinterpret_cast<QosPolicyCountSeq_s*>(qos_policy_countseq);
-                        incompatible_qos_status.last_policy_id() = dds_entity_status->last_policy_id;
-                        incompatible_qos_status.total_count() =
-                                static_cast<dds::OfferedIncompatibleQosStatus*>(dds_entity_status)->
-                                        total_count;
-
-                        data.incompatible_qos_status(std::move(incompatible_qos_status));
-                        delete dds_entity_status;
-
+                        data.incompatible_qos_status(IncompatibleQoSStatus_s{});
+                        if (!status_queryable_.get_monitoring_status(local_entity_guid, data))
+                        {
+                            EPROSIMA_LOG_ERROR(MONITOR_SERVICE,
+                                    "Could not retrieve the incompatible qos entity status ");
+                            status_retrieved = false;
+                            assert(false);
```

Comment:
Why is the assert included here? If this should never happen, then what should be asserted is the condition above. Though I see that this is following the same logic from other cases. Maybe, as this is going to be refactored, it could be taken into account.

Reply:
This assert is already in `master` and `3.0.x-devel`. Out of scope.

Reply:
@EduPonz FYI

Reply:
Yes, we introduced it because it is a behavior that shall never occur, thats the reason why its there

Reply:
In that case, I suggest removing this log and asserting the condition above. This would apply to any other similar code.

```
                            assert(status_queryable_.get_monitoring_status(local_entity_guid, data));
```

Reply:
Its fine for me

Reply:
Applied [here](https://github.com/eProsima/Fast-DDS/pull/4651/commits/e21ae540b2e946f25cdd6a278f100da59f76b0ec) in `master`